File: rust_codeforces/cpp/luogu/old/P1116.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace luogu\P1116.cpp
 *@data      2024/01/17 21:37:06
 */
#include<bits/stdc++.h>
#define ll long long

using namespace std;

void solve() {
    int n, ans = 0;
    cin >> n;
    vector<int>qwq(n);
    for(auto &x : qwq)cin >> x;
    for(int i = 0;i < n;++i)
        for(int j = 0;j < i;++j)
            if(qwq[i] < qwq[j])++ans;
    cout << ans;
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_luogu/src/bin/old/p1116.rs
//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace luogu/p1116.rs
//@data      2024/01/20 14:56:46
fn cin() -> String {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    input.trim().to_string()
}
fn main() {
    let t = 1;
    //let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| solve());
}
fn solve() {
    let n = cin().parse::<usize>().unwrap();
    let mut qwq = cin()
        .split_whitespace()
        .map(|s| s.parse().unwrap())
        .collect::<Vec<i32>>();
    while qwq.len() < n {
        qwq.extend(cin().split_whitespace().map(|s| s.parse::<i32>().unwrap()))
    }
    let ans = (0..n).fold(0, |x, i| {
        x + (0..i).fold(0, |p, j| p + if qwq[i] < qwq[j] { 1 } else { 0 })
    });
    println!("{}", ans);
}
